File: Source/TWXProxy-C#/FormHistory.cs
using System;
using System.Windows.Forms;
using System.Collections.Specialized;
using Core;
 // 
 // Copyright (C) 2005  Remco Mulder
 // 
 // This program is free software; you can redistribute it and/or modify
 // it under the terms of the GNU General Public License as published by
 // the Free Software Foundation; either version 2 of the License, or
 // (at your option) any later version.
 // 
 // This program is distributed in the hope that it will be useful,
 // but WITHOUT ANY WARRANTY; without even the implied warranty of
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 // GNU General Public License for more details.
 // 
 // You should have received a copy of the GNU General Public License
 // along with this program; if not, write to the Free Software
 // Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 // 
 // For source notes please refer to Notes.txt
 // For license terms please refer to GPL.txt.
 // 
 // These files should be stored in the root of the compression you
 // received this source in.
namespace FormHistory
{
    public partial class TfrmHistory: Form
    {
        public TfrmHistory()
        {
            InitializeComponent();
        }

        public void FormShow(Object Sender)
        {
            // centre on screen
            //@ Unsupported property or method(C): 'Width'
            this.Left = (Screen.PrimaryScreen.Width / 2) - (this.Width / 2);
            //@ Unsupported property or method(C): 'Height'
            this.Top = (Screen.PrimaryScreen.Height / 2) - (this.Height / 2);
            clrHistory.Color = memMsgHistory.BackColor;
            fntHistory.Font = memMsgHistory.Font;
        }

        public void btnCloseClick(System.Object Sender, System.EventArgs _e1)
        {
            this.Close();
        }

        public void FormResize(System.Object Sender, System.EventArgs _e1)
        {
            btnClose.Left = this.Width - btnClose.Width - 11;
            btnClear.Left = btnClose.Left - btnClear.Width - 2;
        }

        public void miFontClick(System.Object Sender, System.EventArgs _e1)
        {
            if ((fntHistory.ShowDialog()))
            {
                memFigHistory.Font = fntHistory.Font;
                memComHistory.Font = fntHistory.Font;
                memMsgHistory.Font = fntHistory.Font;
            }
        }

        public void miColourClick(System.Object Sender, System.EventArgs _e1)
        {
            if ((clrHistory.ShowDialog()))
            {
                memFigHistory.BackColor = clrHistory.Color;
                memComHistory.BackColor = clrHistory.Color;
                memMsgHistory.BackColor = clrHistory.Color;
            }
        }

        public void rbClick(System.Object Sender, System.EventArgs _e1)
        {
            if ((rbMessages.Checked))
            {
                memMsgHistory.Visible = true;
                memComHistory.Visible = false;
                memFigHistory.Visible = false;
            }
            else if ((rbFighters.Checked))
            {
                memMsgHistory.Visible = false;
                memComHistory.Visible = false;
                memFigHistory.Visible = true;
            }
            else if ((rbComputer.Checked))
            {
                memMsgHistory.Visible = false;
                memComHistory.Visible = true;
                memFigHistory.Visible = false;
            }
        }

        public void btnClearClick(System.Object Sender, System.EventArgs _e1)
        {
            string ConfirmMessage;
            if ((rbMessages.Checked))
            {
                ConfirmMessage = "message";
            }
            else if ((rbFighters.Checked))
            {
                ConfirmMessage = "fighter";
            }
            else if ((rbComputer.Checked))
            {
                ConfirmMessage = "computer";
            }
            if ((MessageBox.Show("Clear " + ConfirmMessage + " history?", Application.ProductName, System.Windows.Forms.MessageBoxButtons.YesNo| System.Windows.Forms.MessageBoxButtons.YesNo, System.Windows.Forms.MessageBoxIcon.Question) == System.Windows.Forms.DialogResult.Yes))
            {
                if ((rbMessages.Checked))
                {
                    memMsgHistory.Clear();
                }
                else if ((rbFighters.Checked))
                {
                    memFigHistory.Clear();
                }
                else if ((rbComputer.Checked))
                {
                    memComHistory.Clear();
                }
            }
        }

        // Public declarations
        public void AddToHistory(THistoryType HistoryType, string Value)
        {
            StringDictionary Strings;
            switch(HistoryType)
            {
                case Core.THistoryType.htFighter:
                    Strings = memFigHistory.Lines;
                    break;
                case Core.THistoryType.htComputer:
                    Strings = memComHistory.Lines;
                    break;
                case Core.THistoryType.htMsg:
                    Strings = memMsgHistory.Lines;
                    break;
                default:
                    return;
                    break;
            // Shouldn't happen.
            }
            if ((Strings.Count >= 65533))
            {
                Strings.Clear();
            }
            Strings.Add(Value);
        }

    } // end TfrmHistory

}

File: Source/TWX26/FormHistory.pas
{
Copyright (C) 2005  Remco Mulder

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA

For source notes please refer to Notes.txt
For license terms please refer to GPL.txt.

These files should be stored in the root of the compression you 
received this source in.
}
unit FormHistory;

interface

uses
  Core,
  Windows,
  Messages,
  SysUtils,
  Classes,
  Graphics,
  Controls,
  Forms,
  Dialogs,
  StdCtrls,
  ExtCtrls,
  Menus,
  ComCtrls;

type
  TfrmHistory = class(TForm)
    memMsgHistory: TMemo;
    pnlHistory: TPanel;
    rbMessages: TRadioButton;
    btnClose: TButton;
    mnuHistory: TMainMenu;
    View1: TMenuItem;
    miFont: TMenuItem;
    fntHistory: TFontDialog;
    clrHistory: TColorDialog;
    miColour: TMenuItem;
    rbFighters: TRadioButton;
    rbComputer: TRadioButton;
    memFigHistory: TMemo;
    memComHistory: TRichEdit;
    btnClear: TButton;
    procedure FormShow(Sender: TObject);
    procedure btnCloseClick(Sender: TObject);
    procedure FormResize(Sender: TObject);
    procedure miFontClick(Sender: TObject);
    procedure miColourClick(Sender: TObject);
    procedure rbClick(Sender: TObject);
    procedure btnClearClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
    procedure AddToHistory(HistoryType: THistoryType; const Value: string);
  end;

implementation

{$R *.DFM}

procedure TfrmHistory.FormShow(Sender: TObject);
begin
  // centre on screen
  Left := (Screen.Width DIV 2) - (Width DIV 2);
  Top := (Screen.Height DIV 2) - (Height DIV 2);

  clrHistory.Color := memMsgHistory.Color;
  fntHistory.Font := memMsgHistory.Font;
end;

procedure TfrmHistory.btnCloseClick(Sender: TObject);
begin
  Close;
end;

procedure TfrmHistory.FormResize(Sender: TObject);
begin
  btnClose.Left := Width - btnClose.Width - 11;
  btnClear.Left := btnClose.Left - btnClear.Width - 2;
end;

procedure TfrmHistory.miFontClick(Sender: TObject);
begin
  if (fntHistory.Execute) then
  begin
    memFigHistory.Font := fntHistory.Font;
    memComHistory.Font := fntHistory.Font;
    memMsgHistory.Font := fntHistory.Font;
  end;
end;

procedure TfrmHistory.miColourClick(Sender: TObject);
begin
  if (clrHistory.Execute) then
  begin
    memFigHistory.Color := clrHistory.Color;
    memComHistory.Color := clrHistory.Color;
    memMsgHistory.Color := clrHistory.Color;
  end;
end;

procedure TfrmHistory.rbClick(Sender: TObject);
begin
  if (rbMessages.Checked) then
  begin
    memMsgHistory.Visible := TRUE;
    memComHistory.Visible := FALSE;
    memFigHistory.Visible := FALSE;
  end
  else if (rbFighters.Checked) then
  begin
    memMsgHistory.Visible := FALSE;
    memComHistory.Visible := FALSE;
    memFigHistory.Visible := TRUE;
  end
  else if (rbComputer.Checked) then
  begin
    memMsgHistory.Visible := FALSE;
    memComHistory.Visible := TRUE;
    memFigHistory.Visible := FALSE;
  end;
end;

procedure TfrmHistory.btnClearClick(Sender: TObject);
var
  ConfirmMessage : string;
begin
  if (rbMessages.Checked) then
    ConfirmMessage := 'message'
  else if (rbFighters.Checked) then
    ConfirmMessage := 'fighter'
  else if (rbComputer.Checked) then
    ConfirmMessage := 'computer';

  if (MessageDlg('Clear ' + ConfirmMessage + ' history?', mtConfirmation, [mbYes, mbNo], 0) = mrYes) then
  begin
    if (rbMessages.Checked) then
      memMsgHistory.Clear
    else if (rbFighters.Checked) then
      memFigHistory.Clear
    else if (rbComputer.Checked) then
      memComHistory.Clear;
  end;
end;

procedure TfrmHistory.AddToHistory(HistoryType: THistoryType; const Value: string);
var
  Strings: TStrings;
begin
  case HistoryType of
    htFighter  : Strings := memFigHistory.Lines;
    htComputer : Strings := memComHistory.Lines;
    htMsg      : Strings := memMsgHistory.Lines;
  else
    Exit; // Shouldn't happen.
  end;

  if (Strings.Count >= 65533) then
    Strings.Clear;

  Strings.Add(Value);
end;

end.
